.circleci/config.yml

```diff
@@ -18,15 +18,14 @@ commands:
           name: Install Project Dependencies
           command: |
             uv venv
+            source .venv/bin/activate
             # Install root project dependencies
-            uv pip install -e ".[dev]"
+            uv sync --extra dev
             # Install component dependencies
-            cd src/calculator && uv pip install -e ".[dev]"
-            cd ../logger && uv pip install -e ".[dev]"
-            cd ../notifier && uv pip install -e ".[dev]"
+            cd src/calculator && uv sync --extra dev
+            cd ../logger && uv sync --extra dev
+            cd ../notifier && uv sync --extra dev
```

Comment:
why are these needed?
if the are members of the same uv workspace the single uv sync should do?